src/main/scala/scorex/core/network/peer/PeerManager.scala

```diff
@@ -180,7 +207,7 @@ object PeerManager {
     case object GetBlacklistedPeers
 
     // peerCycle messages
-    case class DoConnecting(remote: InetSocketAddress, direction: ConnectionType)
+    case class DoConnecting(remote: InetSocketAddress, direction: ConnectionType) // Only PeerHandler should send this
```

Comment:
I wonder if this kind of restrictions could be enforced/checked in some way by the compiler

@ceilican @kushti 
any ideas on how we could do this?

Reply:
The only idea that comes to my mind is to use `protected[<PeerHandler's package...>]`, but this would require putting PeerHandler in its own package.

Reply:
yes, you're right. But at least we can have a warn comment here.